extensions/sockets/src/main/java/org/mule/module/socket/api/client/TcpListenerClient.java

```diff
@@ -0,0 +1,183 @@
+    {
+        super(protocol, host, port);
+        this.socketProperties = socketProperties;
+        configureSocket();
+    }
+
+    public void configureSocket() throws ConnectionException
+    {
+        try
+        {
+            this.socket = new ServerSocket();
+
+            if (socketProperties.getTimeout() != null)
+            {
+                socket.setSoTimeout(socketProperties.getTimeout());
+            }
+
+            if (socketProperties.getReceiveBufferSize() != null)
+            {
+
+                socket.setReceiveBufferSize(socketProperties.getReceiveBufferSize());
+            }
+
+            if (socketProperties.getReuseAddress() != null)
+            {
+
+                socket.setReuseAddress(socketProperties.getReuseAddress());
+            }
+        }
+        catch (SocketException e)
+        {
+            throw new ConnectionException("Could not configure socket");
+        }
+        catch (IOException e)
+        {
+            throw new ConnectionException("Could not create socket");
+        }
+
+        try
+        {
+            socket.bind(new InetSocketAddress(host, port), socketProperties.getReceiveBacklog());
+        }
+        catch (IOException e)
+        {
+            throw new ConnectionException(String.format("Could not bind socket to host '%s' and port '%d'", host, port));
+        }
+    }
+
+    private Socket listen() throws ConnectionException
+    {
+        try
+        {
+            // blocking
+            Socket newConnection = socket.accept();
+            configureIncomingConnection(newConnection);
+            return newConnection;
+        }
+        catch (SocketTimeoutException e)
+        {
+            // do nothing because timeout is configurable
+            LOGGER.debug("Socket timed out while listening for a new connection");
+            return null;
+        }
+        catch (IOException e)
+        {
+            // this is the expected behaviour while closing the socket
+            return null;
+        }
+        catch (ConnectionException e)
+        {
+            throw e;
+        }
+    }
+
+    private void configureIncomingConnection(Socket newConnection) throws ConnectionException
+    {
+        try
+        {
+            if (socketProperties.getTimeout() != null)
+            {
+                newConnection.setSoTimeout(socketProperties.getTimeout());
+            }
+        }
+        catch (SocketException e)
+        {
+            throw new ConnectionException("Could not configure incoming connection");
+        }
+    }
+
+    public void disconnect() throws ConnectionException
+    {
+        try
+        {
+            if (!socket.isClosed())
+            {
+                socket.close();
+            }
+        }
+        catch (IOException e)
+        {
+            e.printStackTrace();
+            throw new ConnectionException("An error ocurred when closing the listener socket");
```

Comment:
you're not going to reconnect if you fail to close a connection. Just log it as error and move on